simplecnf.go

```diff
@@ -0,0 +1,160 @@
+package go_sat_solver
+
+import "math"
+
+type orClause []VariableAssignment
+
+type simpleSatCnf struct {
+	clauses []orClause
+}
+
+func (f simpleSatCnf) IsSatisfied() bool {
+	return len(f.clauses) == 0
+}
+
+func (f simpleSatCnf) IsContradiction() bool {
+	for _, c := range f.clauses {
+		if len(c) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+type variableAssignment map[Variable]bool
+
+// putAll adds all the assignments to the map
+func (t variableAssignment) putAll(toAdd map[Variable]bool) {
+	m := map[Variable]bool(t)
+	for k, v := range toAdd {
+		m[k] = v
+	}
+}
+
+func (f simpleSatCnf) UnitPropagate() (CnfFormula, map[Variable]bool, []Variable) {
+	assignments := variableAssignment(make(map[Variable]bool))
+	formula := f
+	for {
+		var a map[Variable]bool
+		var conflicts []Variable
+		formula, a, conflicts = formula.unitPropagate1()
+		if len(a) == 0 || len(conflicts) != 0 {
```

Comment:
do you have methods defined somewhere which make it clear what these conditions are? e.g. are these contradictions or complete assignments?